src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -118,67 +144,55 @@ public ProgramEntity createProgram(@NonNull Program program)
 
     val cancers = cancerRepository.findAllByNameIn(program.getCancerTypesList());
     val primarySites = primarySiteRepository.findAllByNameIn(program.getPrimarySitesList());
+    val institutions = institutionRepository.findAllByNameIn(program.getInstitutionsList());
+    val regions = regionRepository.findAllByNameIn(program.getRegionsList());
+    val countries = countryRepository.findAllByNameIn(program.getCountriesList());
```

Comment:
A good example is here, where there is nothing to handle the case of returning an empty list. 

This should make a defensive check on whether or not valid countries were added. 


Something as simple as `countries.size() == program.getCountriesList().size()` could tell you if you had invalid countries entered.